contracts/WamplLoanRouter.sol

```diff
@@ -0,0 +1,110 @@
+pragma solidity ^0.8.3;
+
+import "./interfaces/ILoanRouter.sol";
+import "./interfaces/IBondController.sol";
+import "./interfaces/ITranche.sol";
+import "./interfaces/IButtonWrapper.sol";
+import "./interfaces/IWAMPL.sol";
+import "./interfaces/IWamplLoanRouter.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+/**
+ * @dev Wampl Loan Router built on top of a LoanRouter of your choosing
+ * to allow loans to be created with raw ampl instead of WAMPL
+ */
+contract WamplLoanRouter is IWamplLoanRouter {
+    ILoanRouter public immutable loanRouter;
+    IWAMPL public immutable wampl;
+    IERC20 public immutable ampl;
+
+    /**
+     * @dev Constructor for setting underlying loanRouter and wampl contracts
+     * @param _loanRouter The underlying loanRouter that does all the wrapping and trading
+     * @param _wampl The WAMPL contract to wrap AMPL in
+     */
+    constructor(ILoanRouter _loanRouter, IWAMPL _wampl) {
+        loanRouter = _loanRouter;
+        wampl = _wampl;
+        // Accessing AMPL contract from WAMPL contract
+        ampl = IERC20(_wampl.underlying());
+    }
+
+    /**
+     * @inheritdoc IWamplLoanRouter
+     */
+    function wrapAndBorrow(
+        uint256 underlyingAmount,
```

Comment:
The `underling` makes sense in the context of the wrapper. I dont think its very clear when you use it here in the router. `amplAmt` or `wamplAmt` is probably clearer.